File: examples/typescript/solana/importAccount.ts
// Usage: pnpm tsx evm/importAccount.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import "dotenv/config";
import { Keypair } from "@solana/web3.js";
import bs58 from 'bs58';

const cdp = new CdpClient();

// Importing account with 64-byte private key
console.log("--------------------------------");
console.log("Importing account with 64-byte private key...");
const keypair = Keypair.generate();
const privateKey = bs58.encode(keypair.secretKey); // secretKey is 64 bytes (32 bytes private + 32 bytes public)

const account = await cdp.solana.importAccount({
  privateKey: privateKey,
  name: "MyAccount",
});

console.log("Imported account (64-byte key):", account.address);

// Verify the imported key length
const keyBytes64 = bs58.decode(privateKey);
console.log("Original private key length:", keyBytes64.length, "bytes");

// Importing account with 32-byte private key
console.log("--------------------------------");
console.log("Importing account with 32-byte private key...");
const secondKeypair = Keypair.generate();
const privateKey32 = bs58.encode(secondKeypair.secretKey.subarray(0, 32)); // Take only first 32 bytes (private key only)

const secondAccount = await cdp.solana.importAccount({
  privateKey: privateKey32,
  name: "MyAccount2",
});

console.log("Imported account (32-byte key):", secondAccount.address);

// Verify the imported key length
const keyBytes32 = bs58.decode(privateKey32);
console.log("Original private key length:", keyBytes32.length, "bytes");

console.log("--------------------------------");
console.log("Both accounts imported successfully!");
console.log("64-byte account address:", account.address);
console.log("32-byte account address:", secondAccount.address);

File: examples/python/solana/import_account.py
import asyncio
import base58
from cdp import CdpClient
from dotenv import load_dotenv
from solders.keypair import Keypair


async def main():
     async with CdpClient() as cdp:
        keypair = Keypair()
        
        # Import using base58 encoded private key
        print("1. Importing account using base58 encoded private key:")
        private_key_base58 = base58.b58encode(keypair.to_bytes()).decode('utf-8')
        print(f"Private key (base58): {private_key_base58}")
        
        account1 = await cdp.solana.import_account(
            private_key=private_key_base58,
        )
        print(f"Account imported successfully: {account1.address}\n")
        
        # Import using 32 byte private key array
        print("2. Importing account using 32 byte private key array:")
        new_keypair = Keypair()
        private_key_32_bytes = new_keypair.to_bytes()[:32]  # Take first 32 bytes (seed)
        print(f"Private key length: {len(private_key_32_bytes)} bytes")
        print(f"Private key (hex): {private_key_32_bytes.hex()}")
        
        account2 = await cdp.solana.import_account(
            private_key=private_key_32_bytes,
        )
        print(f"Account imported successfully: {account2.address}\n")
        
        # Verify all accounts have the same address (since they use the same private key for first 3)
        print("=== Verification ===")
        print(f"Account 1 address (base58 string): {account1.address}")
        print(f"Account 2 address (32-byte raw):   {account2.address}")
        
        if account1.address == account2.address:
            print("✓ First two accounts have the same address (as expected)")
        else:
            print("✗ Address mismatch detected")
            
asyncio.run(main())